lib/tools/system-calls.js

```diff
@@ -771,6 +772,21 @@ systemCallMethods.waitForDevice = async function (appDeviceReadyTimeout = 30) {
   });
 };
 
+systemCallMethods.privilegedShell = async function (args) {
+  try {
+    await this.shell(args);
+  } catch (err) {
+    if (err.message.includes('must be root') === -1) {
+      throw err;
+    }
+    // this device needs adb to be running as root to run the specific command.
+    // so try to restart the daemon
+    log.debug(`Device requires adb to be running as root in order to run "adb shell ${args.join(",")}". Restarting daemon`);
+    await this.root();
+    await this.shell(args);
+  }
```

Comment:
how about unrooting in final block?

Reply:
@mykola-mokhnach removed the `unroot` from this method since trying to unroot during a restart makes the code to fail since the device goes offline for a bit.